components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -35,6 +35,22 @@ pub fn date_format(
     Ok(Some(t.unwrap().into_bytes()))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn week_day(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Int>> {
+    if t.is_none() {
+        return Ok(None);
+    }
+    let t = t.as_ref().unwrap();
+    if t.is_zero() {
```

Comment:
If we have a `zero-day` or `zero-month` date, it will return `NULL`.

Reply:
@iosmanthus Thanks, changed to `invalid_zero`.